homeassistant/components/stream/__init__.py

```diff
@@ -272,11 +276,11 @@ def add_provider(
         if not (provider := self._outputs.get(fmt)):
 
             @callback
-            def idle_callback() -> None:
+            def idle_callback(hass: HomeAssistant) -> None:
                 if (
                     not self.keepalive or fmt == RECORDER_PROVIDER
                 ) and fmt in self._outputs:
-                    self.remove_provider(self._outputs[fmt])
+                    hass.add_job(self.remove_provider, self._outputs[fmt])
```

Comment:
We should use the async api inside async context.
```suggestion
                    hass.async_create_task(self.remove_provider, self._outputs[fmt])
```

Reply:
This change seems to be causing the tests to fail with `RuntimeError: Non-thread-safe operation invoked on an event loop other than the current one`, but I'm not sure why. This idle_callback gets called from within async_call_later which should be in the main event loop, no?

Reply:
The fire method isn't decorated with `@callback`,so it will be scheduled on the executor.

https://github.com/home-assistant/core/blob/e5b447839af38d5c10e3205ef095aa205cb6dd16/homeassistant/components/stream/core.py#L222

Reply:
Good catch, thanks!